packages/wonder-blocks-data/components/intercept-data.md

```diff
@@ -0,0 +1,163 @@
+Sometimes, you may want to generate tests that check the loading state and
+subsequent loaded state are working correctly for your uses of `Data`.
+So, rather than manipulating cache usage with `InterceptCache` you can
+manipulate request fulfillment and cache refresh using the `InterceptData`
+component.
+
+This component takes five props; children to be rendered, the handler of the
+type of data requests that are to be intercepted, and either a `fulfillRequest`
+method, a `shouldRefreshCache` method, or both.
+
+Note that this component is expected to be used only within test cases and
+usually only as a single instance. In flight requests for a given handler
+type can be shared and as such, using `InterceptData` alongside non-intercepted
+`Data` components with the same handler type can have undetermined outcomes.
+
+The `fulfillRequest` intercept function has the form:
+
+```js static
+(options: TOptions) => ?Promise<TData>;
+```
+
+If this method is omitted or returns `null`, the default behavior occurs. This
+means that a request will be made for data via the handler assigned to the
+`Data` component being intercepted.
+
+```jsx
+import {Body, BodyMonospace} from "@khanacademy/wonder-blocks-typography";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {InterceptData, Data, RequestHandler} from "@khanacademy/wonder-blocks-data";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+class MyHandler extends RequestHandler {
+    constructor() {
+        super("INTERCEPT_DATA_HANDLER1");
+    }
+
+    fulfillRequest(options) {
+        return Promise.reject(new Error("You should not see this!"));
+    }
+}
+
+const handler = new MyHandler();
+const fulfillRequestInterceptor = function(options) {
+    if (options === "DATA") {
+        return Promise.resolve("INTERCEPTED DATA!");
+    }
+    return null;
+};
+
+<InterceptData handler={handler} fulfillRequest={fulfillRequestInterceptor}>
+    <View>
+        <Body>This received intercepted data!</Body>
+        <Data handler={handler} options={"DATA"}>
```

Comment:
The handler needs to be the same for these two otherwise the `InterceptData` won't override `Data`'s behavior, right?

Reply:
Yup.